metadata-service/auth-filter/src/test/java/AuthenticationFilterTest.java

```diff
@@ -0,0 +1,54 @@
+import com.datahub.auth.authentication.filter.AuthenticationFilter;
+import com.datahub.authentication.Actor;
+import com.datahub.authentication.ActorType;
+import com.datahub.authentication.token.StatefulTokenService;
+import com.datahub.authentication.token.TokenClaims;
+import com.datahub.authentication.token.TokenException;
+import com.datahub.authentication.token.TokenType;
+import com.datahub.authentication.token.TokenVersion;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.Test;
+
+import static com.datahub.authentication.AuthenticationConstants.*;
+import static org.mockito.Mockito.*;
+
+
+@ContextConfiguration(classes = { AuthTestConfiguration.class })
+public class AuthenticationFilterTest extends AbstractTestNGSpringContextTests {
+
+  @Autowired
+  AuthenticationFilter _authenticationFilter;
+
+  @Autowired
+  StatefulTokenService _statefulTokenService;
+
+  @Test
+  public void testExpiredToken() throws ServletException, IOException, TokenException {
+    _authenticationFilter.init(null);
+    HttpServletRequest servletRequest = mock(HttpServletRequest.class);
+    HttpServletResponse servletResponse = mock(HttpServletResponse.class);
+    FilterChain filterChain = mock(FilterChain.class);
+    Actor actor = new Actor(ActorType.USER, "datahub");
+//    String token = _statefulTokenService.generateAccessToken(TokenType.SESSION, actor, 0L, System.currentTimeMillis(), "token",
```

Comment:
Why is this commented out? If it's not used might as well delete. 
Is there no need to test the stateful token service?

Reply:
This shows how the token was generated, I left in to make it more obvious

Reply:
It's not super intuitive to know how the token is generated otherwise so I'd prefer to leave it in for anyone who wants to make a similar test and looks through this file if it doesn't bother you too much :)